site/app/templates/NotificationSettings.twig

```diff
@@ -115,10 +119,9 @@
                         </label>
                     </div>
                     <div class="option-input col-md-3"><input type="checkbox" name="all_modifications_forum" id="all_modifications_forum" {% if notification_saves['all_modifications_forum'] %}checked="true"{% endif %}></div>
-
-                    <div class="option-input col-md-5"><input type="checkbox" name="all_modifications_forum_email" id="all_modifications_forum_email" {% if notification_saves['all_modifications_forum_email'] %}checked="true"{% endif %}></div>
-
-
+                    {% if email_enabled %}
+                        <div class="option-input col-md-5"><input type="checkbox" name="all_modifications_forum_email" id="all_modifications_forum_email" {% if notification_saves['all_modifications_forum_email'] %}checked="true"{% endif %}></div>
+                    {% endif %}
```

Comment:
We should probably update `NotificationController::saveSettings` to not overwrite values that are missing from the page here as it can lead to some unexpectedness if email is enabled for a while, users set their settings turning on email notifications, email is disabled, user goes to the page and disables a setting (all email settings are now turned off), and then email is re-enabled.

Reply:
The user shouldnt be able to change their email settings if emails are disabled as that column to edit email settings would not be shown to them. However it is probably safe to have that check.